src/pypi_attestations/_impl.py

```diff
@@ -34,6 +34,30 @@
     from sigstore.verify.policy import VerificationPolicy  # pragma: no cover
 
 
+class Base64EncoderSansNewline(Base64Encoder):
+    r"""A Base64Encoder that doesn't insert newlines when encoding.
+
+    Pydantic's Base64Bytes type inserts newlines b'\n' every 76 characters because they
+    use `base64.encodebytes()` instead of `base64.b64encode()`. Pydantic maintainers
+    have stated that they won't fix this, and that users should work around it by
+    defining their own Base64 type with a custom encoder.
+    See https://github.com/pydantic/pydantic/issues/9072 for more details.
+    """
+
+    @classmethod
+    def encode(cls, value: bytes) -> bytes:
+        """Encode bytes to base64."""
+        return base64.b64encode(value)
+
+    @classmethod
+    def decode(cls, value: bytes) -> bytes:
+        """Decode base64 bytes."""
+        return base64.b64decode(value, validate=True)
+
+
+Base64Bytes = Annotated[bytes, EncodedBytes(encoder=Base64EncoderSansNewline)]
```

Comment:
Let's add a backstop test for this type as well, to assert that it absolutely never contains newlines in its encodings and fails when newlines are present while decoding.

Reply:
done!